tools/package-source.js

```diff
@@ -99,6 +99,93 @@ var splitConstraint = function (c) {
            constraint: parsed.constraintString || null };
 };
 
+// Given the text of a README.md file, excerpts the text between the first and
+// second heading.
+//
+// Specifically - if there is text between the document name, and the first
+// subheading, it will take that text. If there is no text there, and only text
+// after the first subheading, it will take that text. It won't look any deeper
+// than that (in case the user intentionally wants to leave the section blank
+// for some reason). Skips lines that start with an exclamation point.
+var getExcerptFromReadme = function (text) {
+  // Split into lines.
+  var fullText = text.split("\n");
+  if (! fullText) {
+    return "";
+  }
+
+  // The lines of the excerpt that we will return to the caller.
+  var excerpt = [];
+
+  // We want to take the data between the first and second headers, so it is
+  // important to keep track of how many headers we have passed.
+  var headerNum = 0;
+
+  // Counter and termination condition for the while loop.
+  var lineNum = 0;
+  var done = false;
+
+  // We want to scan until we have an excerpt, or decide that we are not getting
+  // one. If we are only taking the first five lines of a long README file, we
+  // want to avoid scanning all of it.
+  while (! done) {
+    var currentLine = fullText[lineNum];
+    var trimmedLine = currentLine.trim();
+
+    // According to "https://help.github.com/articles/markdown-basics/", a
+    // heading is denoted by a '#' symbol, though, I think, the line at the
+    // top (===) should also count as a header for our purposes.
+    var isHeader =
+      (trimmedLine[0] === "#") ||
+      (trimmedLine[0] === "=");
```

Comment:
I am actually only responding to this comments to basically whine that I have used commonmark for a day, and it is already failing some of my most basic tests. (Because, sometimes, a block's `end_line` is less than its `start_line`, which according David 'seems like a bug'). I have also tried some of the other common NPM modules, and ended up breaking them pretty easily. (Maybe my Markdown was corrupt though, I am not sure.) I am less concerned about this particular bug, than I am that it didn't take me very long to run into it.

My other thought is that we don't need a lot of capability, at least not yet. Parsing the Markdown introduces parsing potential errors that a most basic implementation can avoid. That's probably good, since we just want to grab the basic raw contents*.

Parsing out some headers out of Markdown seemed pretty easy to do manually, and actually less prone to stupid errors. Also, I figured that it was probably easier to explain to users -- "We take things between the hashtag lines" is a lot more clear than "we take things between the headers unless <npm package name> is having a bad day or something". Though, on the other hand, I agree that re-doing the work that 50 other people, including David, have done is super annoying. So, IDK. 

I will probably wrestle with commonmark some more, but wanted to put this out there.

* The idea was that, for now at least, show won't really process Markdown beyond taking a snippet. We are only asking for a small paragraph, and processing Markdown can lead down a long, twisty path. Plus, Markdown is intentionally meant to be human-legible. (So, while it sucks that some links won't render great, it is probably OK.)

We might want to parse things some day, but we might also want to have a more nuanced control file, or a 'meteor docs' command that opens up an app and renders Markdown as something pretty. So, it seemed reasonable to leave it out of the first version. (Then again, if we can get it for free, maybe it is worth it.)